generate_e2e_test.go

```diff
@@ -97,6 +97,105 @@ func TestRunGenerate(t *testing.T) {
 	}
 }
 
+func TestCleanAndCopy(t *testing.T) {
+	const (
+		localAPISource = "testdata/e2e/generate/api_root"
+		apiToGenerate  = "google/cloud/pubsub/v1"
+	)
+	// create a temp directory for writing files, so we don't have to create testdata files.
+	repoInitDir := t.TempDir()
+	// create a file that should be removed.
+	if err := os.WriteFile(filepath.Join(repoInitDir, "file_to_remove.txt"), []byte("remove me"), 0644); err != nil {
+		t.Fatal(err)
+	}
+	// create a directory with 2 files, on of them should be preserved.
+	if err := os.MkdirAll(filepath.Join(repoInitDir, "sub"), 0755); err != nil {
+		t.Fatal(err)
+	}
+	if err := os.WriteFile(filepath.Join(repoInitDir, "sub", "file_to_preserve.txt"), []byte("preserve me"), 0644); err != nil {
+		t.Fatal(err)
+	}
+	if err := os.WriteFile(filepath.Join(repoInitDir, "sub", "file_to_remove_in_sub.txt"), []byte("remove me"), 0644); err != nil {
+		t.Fatal(err)
+	}
+	// create a state file with remove and preserve regex.
+	state := &config.LibrarianState{
+		Image: "test-image:latest",
+		Libraries: []*config.LibraryState{
+			{
+				ID:      "go-google-cloud-pubsub-v1",
+				Version: "v1.0.0",
+				APIs: []*config.API{
+					{
+						Path: "google/cloud/pubsub/v1",
+					},
+				},
+				SourceRoots: []string{"cloud.google.com/go/pubsub"},
+				RemoveRegex: []string{
+					"file_to_remove.txt",
+					"^sub/.*.txt",
+				},
```

Comment:
Should directories to remove be subdirectories of source roots?

I think, when generating a library, only directories within its source roots should be removed or preserved.